code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,849 @@
+	playsound(src, 'sound/items/welder.ogg', 150, TRUE)
+	qdel(src)
+
+
+///nearly all runes use their turf in some way so we set rune_turf to their turf automatically, the rune also start on cooldown if it uses one.
+/obj/effect/warped_rune/Initialize()
+	. = ..()
+	add_overlay("blank", TRUE)
+	rune_turf = get_turf(src)
+	RegisterSignal(rune_turf, COMSIG_COMPONENT_CLEAN_ACT, .proc/clean_rune)
+
+/obj/effect/warped_rune/proc/clean_rune()
+	qdel(src)
+
+///using the extract on the floor will "draw" the rune.
+/obj/item/slimecross/warping/afterattack(atom/target, mob/user, proximity)
+	. = ..()
+	if(!proximity)
+		return
+
+	if(istype(target, runepath)) //checks if the target is a rune and then if you can store it
+		if(do_after(user, storing_time,target = target))
+			warping_crossbreed_absorb(target, user)
+		return
+
+	if(isturf(target) && locate(/obj/effect/warped_rune) in target) //check if the target is a floor and if there's a rune on said floor
+		to_chat(user, "<span class='warning'>There is already a bluespace rune here!</span>")
+		return
+
+	if(!istype(target,/turf/open/floor))
+		to_chat(user, "<span class='warning'>you cannot draw a rune here!</span>")
+		return
+
+	if(warp_charge < 1) //check if we have at least 1 charge left.
+		to_chat(user, "<span class='warning'>[src] is empty!</span>")
+		return
+
+	if(!check_cd(user))
+		return
+
+	if(do_after(user, drawing_time,target = target))
+		if(warp_charge >= 1 && (!locate(/obj/effect/warped_rune) in target) && check_cd(user)) //check one last time if a rune has been drawn during the do_after and if there's enough charges left
+			warping_crossbreed_spawn(target,user)
+			make_cd()
+
+
+///spawns the rune, taking away one rune charge
+/obj/item/slimecross/warping/proc/warping_crossbreed_spawn(atom/target, mob/user)
+	playsound(target, 'sound/effects/slosh.ogg', 20, TRUE)
+	warp_charge--
+	new runepath(target)
+	to_chat(user, "<span class='notice'>You carefully draw the rune with [src].</span>")
+
+
+///absorb the rune into the crossbreed adding one more charge to the crossbreed.
+/obj/item/slimecross/warping/proc/warping_crossbreed_absorb(atom/target, mob/user)
+	//to_chat(user, "<span class='notice'>You store the rune in [src].</span>")
+	qdel(target)
+	warp_charge++
+
+/obj/item/slimecross/warping/proc/check_cd(user)
+	if(world.time < cooldown)
+		if(user)
+			to_chat(user, "<span class='warning'>[src] is recharging energy.</span>")
+		return FALSE
+	return TRUE
+
+/obj/item/slimecross/warping/proc/make_cd()
+	cooldown = world.time + max_cooldown
+
+/obj/effect/warped_rune/attack_hand(mob/living/user)
+	. = ..()
+	do_effect(user)
+
+/obj/effect/warped_rune/proc/do_effect(mob/user)
+	SHOULD_CALL_PARENT(TRUE)
+	if(deleteme)
+		playsound(rune_turf, dir_sound, 20, TRUE)
+		to_chat(user, ("<span class='notice'>[src] fades.</span>"))
+		qdel(src)
+
+/obj/effect/warped_rune/Crossed(atom/movable/AM, oldloc)
+	SHOULD_CALL_PARENT(TRUE)
```

Comment:
Is this needed? Same with the one above

Reply:
the code can work without this but in the end all the code is oriented towards calling the parent proc. I also don't see what harm it does.